src/dotnet/commands/dotnet-tool/uninstall/ToolUninstallCommand.cs

```diff
@@ -9,131 +9,89 @@
-            IReporter reporter = null)
+            IReporter reporter = null,
+            ToolUninstallGlobalOrToolPathCommand toolUninstallGlobalOrToolPathCommand = null,
+            ToolUninstallLocalCommand toolUninstallLocalCommand = null)
             : base(result)
         {
-            _options = options ?? throw new ArgumentNullException(nameof(options));
-            _reporter = reporter ?? Reporter.Output;
-            _errorReporter = reporter ?? Reporter.Error;
+            _toolUninstallLocalCommand
+                = toolUninstallLocalCommand ??
+                  new ToolUninstallLocalCommand(options, result);
+
+            _toolUninstallGlobalOrToolPathCommand =
+                toolUninstallGlobalOrToolPathCommand
+                ?? new ToolUninstallGlobalOrToolPathCommand(options, result);
 
-            _createShellShimRepository = createShellShimRepository ?? ShellShimRepositoryFactory.CreateShellShimRepository;
-            _createToolPackageStoresAndUninstaller = createToolPackageStoreAndUninstaller ??
-                                                    ToolPackageFactory.CreateToolPackageStoresAndUninstaller;
+            _global = options.ValueOrDefault<bool>(GlobalOption);
+            _local = options.ValueOrDefault<bool>(LocalOption);
+            _toolPath = options.SingleArgumentOrDefault(ToolPathOption);
+            _toolManifestOption = options.ValueOrDefault<string>("tool-manifest");
         }
 
         public override int Execute()
         {
-            var global = _options.ValueOrDefault<bool>("global");
-            var toolPath = _options.SingleArgumentOrDefault("tool-path");
+            EnsureNoConflictGlobalLocalToolPathOption();
 
-            DirectoryPath? toolDirectoryPath = null;
-            if (!string.IsNullOrWhiteSpace(toolPath))
+            if (_global || !string.IsNullOrWhiteSpace(_toolPath))
             {
-                if (!Directory.Exists(toolPath))
+                if (!string.IsNullOrWhiteSpace(_toolManifestOption))
                 {
                     throw new GracefulException(
                         string.Format(
-                            LocalizableStrings.InvalidToolPathOption,
-                            toolPath));
+                            LocalizableStrings.OnlyLocalOptionSupportManifestFileOption));
                 }
 
-                toolDirectoryPath = new DirectoryPath(toolPath);
+                return _toolUninstallGlobalOrToolPathCommand.Execute();
             }
-
-            if (toolDirectoryPath == null && !global)
+            else
             {
-                throw new GracefulException(LocalizableStrings.UninstallToolCommandNeedGlobalOrToolPath);
+                return _toolUninstallLocalCommand.Execute();
             }
+        }
 
-            if (toolDirectoryPath != null && global)
+        private void EnsureNoConflictGlobalLocalToolPathOption()
+        {
+            List<string> options = new List<string>();
```

Comment:
If we do remove listing the "conflicting options" from the error message, I would love to see this method replaced with a simple boolean check rather than pushing items to a list that is only used in the error case.